modules/heat_transfer/src/userobjects/ViewFactorRayStudy.C

```diff
@@ -312,6 +314,68 @@ ViewFactorRayStudy::generateRays()
                                 : std::cos(_2d_aq_angles[l]) * _2d_aq_weights[l];
         const auto start_weight = start_elem._weights[start_i] * awf;
 
+        // Modify the sampled direction to at worst graze the face of 3D protruding non-planar
+        // elements. We do not expect there are any neighbor elements to track in if it exits the
+        // non-planar side.
+        // TODO: modify the weights to keep the quadrature order
+        bool intersection_found = false;
+        if (_is_3d && start_elem._start_elem &&
```

Comment:
We should only call `side_ptr` once; it builds a whole new element. there's a helper that does this to avoid construction